main.go

```diff
@@ -154,20 +160,26 @@ func main() {
 		os.Exit(1)
 	}
 
+	clusterCapabilities := cluster.Capabilities{
+		// etcd quorum PDB is only installed in OpenShift
+		HasEtcdPDBQuorum: onOpenshift,
+		HasMachineAPI:    hasMachineAPI,
+	}
+
 	if err := (&controllers.NodeHealthCheckReconciler{
 		Client:                      mgr.GetClient(),
 		Log:                         ctrl.Log.WithName("controllers").WithName("NodeHealthCheck"),
 		Recorder:                    mgr.GetEventRecorderFor("NodeHealthCheck"),
 		ClusterUpgradeStatusChecker: upgradeChecker,
 		MHCChecker:                  mhcChecker,
-		OnOpenShift:                 onOpenshift,
+		Capabilities:                clusterCapabilities,
 		MHCEvents:                   mhcEvents,
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "NodeHealthCheck")
 		os.Exit(1)
 	}
 
-	if onOpenshift {
+	if onOpenshift && hasMachineAPI {
```

Comment:
hmm isn't ` onOpenshift` redundant ?
or in other words  is there a use case for non-openshift cluster that has Machine API ?

Reply:
Honestly I am not sure :D
What I wanted to do here was to "declare" in advace all that was necessary for the content inside the `if` branch to work

- the feature gate is on OCP (IIUC) -> isOnOpenshit
- the Machine related part needs MachineAPI -> HasMachineAPI

but I don't have strong feeling about this, if you think it's better to remove `onOpenshift` it's ok for me

Reply:
Since I'm not 100% sure myself I really  don't mind keeping `onOpenshift`.
Worst case scenario we'll keep a redundant field that does no harm.